\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={R\_Assignment},
            pdfauthor={Elizabeth Glynne},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{R\_Assignment}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Elizabeth Glynne}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{10/16/2019}


\begin{document}
\maketitle

\emph{RMD for R Assignment: This is the markdown file for my assignment.
It will describe my general workflow for this project.}

\hypertarget{link-r-and-github}{%
\subsubsection{Link R and Github}\label{link-r-and-github}}

I created a repository on Github and then created a new project from
Version control.

\hypertarget{part-i-data-inspection}{%
\subsection{Part I: Data inspection}\label{part-i-data-inspection}}

\hypertarget{step-1-import-data}{%
\subsubsection{Step 1: Import Data}\label{step-1-import-data}}

I copied the data from the UNIX assignment into my R\_Assignment folder
using my local repositories. I then pushed them to my global repository
and then pulled the repository through my Rstudio interface.

Loading in the genotypes data:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(tidyverse)}
\NormalTok{fang <-}\StringTok{ }\KeywordTok{read_tsv}\NormalTok{(}\DataTypeTok{file =} \StringTok{"fang_et_al_genotypes.txt"}\NormalTok{, }\DataTypeTok{col_names =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Loading in the SNPs data:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{snp <-}\StringTok{ }\KeywordTok{read_tsv}\NormalTok{(}\DataTypeTok{file =} \StringTok{"snp_position.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{step-2-inspect-genotypes}{%
\subsubsection{Step 2: Inspect
Genotypes}\label{step-2-inspect-genotypes}}

View the data file:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{View}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

In viewing the data file, we can see that it is a table with headers.

This file is currently a list:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{typeof}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

This file has 986 variables, with 2782 rows and 986 columns.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{length}\NormalTok{(fang)}
\KeywordTok{dim}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

By first storing the attributes of the genotype file as an object, I can
further analyze each attribute of the list.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genotype_attributes <-}\StringTok{ }\KeywordTok{attributes}\NormalTok{(fang)}
\KeywordTok{length}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{names)}
\KeywordTok{length}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{class)}
\KeywordTok{length}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{row.names)}
\KeywordTok{typeof}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{names)}
\KeywordTok{typeof}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{row.names)}
\KeywordTok{typeof}\NormalTok{(genotype_attributes}\OperatorTok{$}\NormalTok{class)}
\NormalTok{genotype_attributes}\OperatorTok{$}\NormalTok{class}
\end{Highlighting}
\end{Shaded}

We can see that the lengths of each attribute, how the names correspond
to the number of columns (986) and the row names correspond to the
number of rows (2782). However, the length of the classes (4)
corresponds to the various components being a special table dataframe,
table dataframe, table, and a data frame.

\hypertarget{step-3-inspect-snps}{%
\subsubsection{Step 3: Inspect SNPs}\label{step-3-inspect-snps}}

View the data file:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{View}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

In viewing the data file, we can see that it is a table with headers.

This file is currently a list:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{typeof}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "list"
\end{verbatim}

This file has 15 variables, with 983 rows and 15 columns of data.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{length}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 15
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{dim}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 983  15
\end{verbatim}

By first storing the attributes of the snps file as an object, I can
further analyze each attribute of the list.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{snp_attributes <-}\StringTok{ }\KeywordTok{attributes}\NormalTok{(snp)}
\KeywordTok{length}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{names)}
\KeywordTok{length}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{class)}
\KeywordTok{length}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{row.names)}
\KeywordTok{typeof}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{names)}
\KeywordTok{typeof}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{row.names)}
\KeywordTok{typeof}\NormalTok{(snp_attributes}\OperatorTok{$}\NormalTok{class)}
\NormalTok{snp_attributes}\OperatorTok{$}\NormalTok{class}
\end{Highlighting}
\end{Shaded}

We can see that the lengths of each attribute, how the names correspond
to the number of columns (15) and the row names correspond to the number
of rows (983). However, the length of the classes (4) corresponds to the
various components being a special table dataframe, table dataframe,
table, and a data frame.

\hypertarget{part-ii-data-processing}{%
\subsection{Part II: Data Processing}\label{part-ii-data-processing}}

\hypertarget{step-1-prep-files}{%
\subsubsection{Step 1: Prep files}\label{step-1-prep-files}}

Sort the SNPs by ID

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(tidyr)}
\NormalTok{sorted_snp<-}\KeywordTok{arrange}\NormalTok{(snp, SNP_ID)}
\end{Highlighting}
\end{Shaded}

\hypertarget{step-1-define-groups}{%
\subsubsection{Step 1: Define groups}\label{step-1-define-groups}}

Pull out Maize (ZMMIL, ZMMLR, and ZMMMR)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize<-}\StringTok{ }\KeywordTok{filter}\NormalTok{(fang, Group }\OperatorTok{%in%}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"ZMMIL"}\NormalTok{, }\StringTok{"ZMMLR"}\NormalTok{, }\StringTok{"ZMMMR"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

Pull out Teosinte (ZMPBA, ZMPIL, ZMPJA)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{teosinte<-}\StringTok{ }\KeywordTok{filter}\NormalTok{(fang, Group }\OperatorTok{%in%}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"ZMPBA"}\NormalTok{, }\StringTok{"ZMPIL"}\NormalTok{, }\StringTok{"ZMPJA"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{step-2-transpose-file-maize-and-teosinte}{%
\subsubsection{Step 2: Transpose file Maize and
Teosinte}\label{step-2-transpose-file-maize-and-teosinte}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Tmaize<-}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{t}\NormalTok{(maize))}
\NormalTok{Tteosinte <-}\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{t}\NormalTok{(teosinte))}
\end{Highlighting}
\end{Shaded}

\hypertarget{step-3-join-snp-and-genotype-files-for-both-maize-and-teosinte}{%
\subsubsection{Step 3: Join SNP and Genotype files for both Maize and
Teosinte}\label{step-3-join-snp-and-genotype-files-for-both-maize-and-teosinte}}

Join together the files into a new table, with the first three columns
from the sorted\_snp file (SNP\_ID, Chromosome, and Position). This will
be used with the creation of the next two files.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{join_start<-}\KeywordTok{bind_cols}\NormalTok{(}\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{SNP_ID =}\NormalTok{ sorted_snp}\OperatorTok{$}\NormalTok{SNP_ID),}\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{Chromosome =}\NormalTok{ sorted_snp}\OperatorTok{$}\NormalTok{Chromosome),}\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{Position =}\NormalTok{ sorted_snp}\OperatorTok{$}\NormalTok{Position))}
\end{Highlighting}
\end{Shaded}

\hypertarget{step-4-create-the-10-chromosome-files-for-maize}{%
\subsubsection{Step 4: Create the 10 chromosome files for
Maize}\label{step-4-create-the-10-chromosome-files-for-maize}}

Change dataframe so the row names are now in a column called SNP\_ID.
Join together the Tmaize file with the join\_start file, resulting in
the join\_maize file which will be used for later.

Create join\_maize1 file, changing ``?/?'' to ``?''. Create join\_maize2
file, changing ``?'' to '-".

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{join_maize <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\KeywordTok{lapply}\NormalTok{(join_maize, as.character),}
                            \DataTypeTok{stringsAsFactors=}\OtherTok{FALSE}\NormalTok{)}
\NormalTok{join_maize1 <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\KeywordTok{sapply}\NormalTok{(join_maize,}\ControlFlowTok{function}\NormalTok{(x) }
\NormalTok{                  \{x <-}\StringTok{ }\KeywordTok{gsub}\NormalTok{(}\StringTok{"?/?"}\NormalTok{,}\StringTok{"?"}\NormalTok{,x,}\DataTypeTok{fixed=}\OtherTok{TRUE}\NormalTok{)\}))}
\NormalTok{join_maize2 <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\KeywordTok{sapply}\NormalTok{(join_maize1,}\ControlFlowTok{function}\NormalTok{(x) }
\NormalTok{                  \{x <-}\StringTok{ }\KeywordTok{gsub}\NormalTok{(}\StringTok{"?"}\NormalTok{,}\StringTok{"-"}\NormalTok{,x,}\DataTypeTok{fixed=}\OtherTok{TRUE}\NormalTok{)\}))}
\end{Highlighting}
\end{Shaded}

\hypertarget{i-create-a-file-for-each-chromosome-with-positions-in-increasing-order-and-missing-data-denoted-as}{%
\paragraph{i: Create a file for each chromosome with positions in
increasing order and missing data denoted as
``?''}\label{i-create-a-file-for-each-chromosome-with-positions-in-increasing-order-and-missing-data-denoted-as}}

Sort the join\_maize file into 10 new data frames, one for each
chromosome. Order the files based on the increasing position values:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{df.names <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(}\StringTok{"maize_chrom"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{,}\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{) \{}
\NormalTok{  d.frame <-}\StringTok{ }\KeywordTok{filter}\NormalTok{(join_maize1, Chromosome }\OperatorTok{==}\StringTok{ }\NormalTok{i)}
  \KeywordTok{assign}\NormalTok{(df.names[i],d.frame)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Order the files based on the increasing postion values:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{1}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{1}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{1}\OperatorTok{$}\NormalTok{Position))),]}
\NormalTok{maize_chrom_}\DecValTok{2}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{2}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{2}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in order(as.numeric(as.character(maize_chrom_2$Position))): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{3}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{3}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{3}\OperatorTok{$}\NormalTok{Position))),]}
\NormalTok{maize_chrom_}\DecValTok{4}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{4}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{4}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in order(as.numeric(as.character(maize_chrom_4$Position))): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{5}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{5}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{5}\OperatorTok{$}\NormalTok{Position))),]}
\NormalTok{maize_chrom_}\DecValTok{6}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{6}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{6}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in order(as.numeric(as.character(maize_chrom_6$Position))): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{7}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{7}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{7}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in order(as.numeric(as.character(maize_chrom_7$Position))): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{8}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{8}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{8}\OperatorTok{$}\NormalTok{Position))),]}
\NormalTok{maize_chrom_}\DecValTok{9}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{9}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{9}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in order(as.numeric(as.character(maize_chrom_9$Position))): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maize_chrom_}\DecValTok{10}\NormalTok{_s<-}\StringTok{ }\NormalTok{maize_chrom_}\DecValTok{10}\NormalTok{[}\KeywordTok{order}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(maize_chrom_}\DecValTok{10}\OperatorTok{$}\NormalTok{Position))),]}
\end{Highlighting}
\end{Shaded}

\hypertarget{ii-create-a-file-for-each-chromosome-with-positions-in-decreasing-order-and-missing-data-denoted-as--}{%
\paragraph{ii: Create a file for each chromosome with positions in
decreasing order and missing data denoted as
``-''}\label{ii-create-a-file-for-each-chromosome-with-positions-in-decreasing-order-and-missing-data-denoted-as--}}

Sort the join\_maize file into 10 new data frames, one for each
chromosome.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{df.names <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(}\StringTok{"maize_chrom_de"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{,}\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{) \{}
\NormalTok{  d.frame <-}\StringTok{ }\KeywordTok{filter}\NormalTok{(join_maize2, Chromosome }\OperatorTok{==}\StringTok{ }\NormalTok{i)}
  \KeywordTok{assign}\NormalTok{(df.names[i],d.frame)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Order the files based on the decreasing position values:

A file with all SNPs with unknown positions in the genome:

A file with all SNPs with multiple positions in the genome:

\hypertarget{step-5-create-the-10-chromosome-files-for-teosinte}{%
\subsubsection{Step 5: Create the 10 chromosome files for
Teosinte}\label{step-5-create-the-10-chromosome-files-for-teosinte}}

Change dataframe so the row names are now in a column called SNP\_ID.
Join together the Tteosinte file with the join\_start file, resulting in
the join\_teosinte file which will be used for later analysis.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Tteosinte<-tibble}\OperatorTok{::}\KeywordTok{rownames_to_column}\NormalTok{(Tteosinte, }\StringTok{"SNP_ID"}\NormalTok{) }
\NormalTok{join_teosinte <-}\StringTok{ }\KeywordTok{full_join}\NormalTok{(join_start, Tteosinte, }\DataTypeTok{by=} \StringTok{"SNP_ID"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Column `SNP_ID` joining factor and character vector, coercing into
## character vector
\end{verbatim}

Sort the join\_teosinte file into 10 new data frames, one for each
chromosome.Order the files based on the increasing position values:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{df.names <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(}\StringTok{"teosinte_chrom"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{,}\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{) \{}
\NormalTok{  d.frame <-}\StringTok{ }\KeywordTok{filter}\NormalTok{(join_teosinte, Chromosome }\OperatorTok{==}\StringTok{ }\NormalTok{i)}
  \KeywordTok{arrange}\NormalTok{(d.frame, Position)    }
  \KeywordTok{assign}\NormalTok{(df.names[i],d.frame)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Change the missing data to ``?'' in the data frame:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(plyr)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: package 'plyr' was built under R version 3.6.1
\end{verbatim}

\begin{verbatim}
## -------------------------------------------------------------------------
\end{verbatim}

\begin{verbatim}
## You have loaded plyr after dplyr - this is likely to cause problems.
## If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
## library(plyr); library(dplyr)
\end{verbatim}

\begin{verbatim}
## -------------------------------------------------------------------------
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'plyr'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:dplyr':
## 
##     arrange, count, desc, failwith, id, mutate, rename, summarise,
##     summarize
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:purrr':
## 
##     compact
\end{verbatim}

Sort the join\_teosinte files to 10 new data frames, one for each
chromosome. Order the files based on the decreasing position values:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{df.names <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(}\StringTok{"teosinte_chrom_de"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{,}\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\DecValTok{10}\NormalTok{) \{}
\NormalTok{  d.frame <-}\StringTok{ }\KeywordTok{filter}\NormalTok{(join_teosinte, Chromosome }\OperatorTok{==}\StringTok{ }\NormalTok{i)}
  \KeywordTok{arrange}\NormalTok{(d.frame, Position, }\DataTypeTok{decreasing =} \OtherTok{FALSE}\NormalTok{)    }
  \KeywordTok{assign}\NormalTok{(df.names[i],d.frame)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Change the missing data from ``?'' to ``-'':

A file with all SNPs with unknown positions in the genome:

A file with all SNPs with multiple positions in the genome:

\hypertarget{part-iii-data-visualization}{%
\subsection{Part III: Data
Visualization}\label{part-iii-data-visualization}}

\hypertarget{snps-per-chromosom}{%
\subsubsection{SNPs per chromosom}\label{snps-per-chromosom}}

\hypertarget{missing-data-and-amount-of-heterozygosity}{%
\subsubsection{Missing data and amount of
heterozygosity}\label{missing-data-and-amount-of-heterozygosity}}

\hypertarget{your-own-visualization}{%
\subsubsection{"Your own visualization}\label{your-own-visualization}}


\end{document}
