---
title: "R_Assignment"
author: "Elizabeth Glynne"
date: "10/16/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
_RMD for R Assignment: This is the markdown file for my assignment. It will describe my general workflow for this project._

### Link R and Github

I created a repository on Github and then created a new project from Version control.

## Part I: Data inspection
### Step 1: Import Data

I copied the data from the UNIX assignment into my R_Assignment folder using my local repositories. I then pushed them to my global repository and then pulled the repository through my Rstudio interface.

Loading in the genotypes data:
```{r message=FALSE, warning=FALSE, results=FALSE}
library(tidyverse)
fang <- read_tsv(file = "fang_et_al_genotypes.txt", col_names = TRUE)
```

Loading in the SNPs data:
```{r warning=FALSE, message=FALSE, results=FALSE}
snp <- read_tsv(file = "snp_position.txt")
```

### Step 2: Inspect Genotypes
View the data file:

```{r}
View(fang)
```
In viewing the data file, we can see that it is a table with headers. 

This file is currently a list:
```{r}
typeof(fang)
```
This file has 986 variables, with 2782 rows and 986 columns.
```{r}
length(fang)
dim(fang)
```
By first storing the attributes of the genotype file as an object, I can further analyze each attribute of the list.  
```{r results=FALSE}
genotype_attributes <- attributes(fang)
length(genotype_attributes$names)
length(genotype_attributes$class)
length(genotype_attributes$row.names)
typeof(genotype_attributes$names)
typeof(genotype_attributes$row.names)
typeof(genotype_attributes$class)
genotype_attributes$class
```
We can see that the lengths of each attribute, how the names correspond to the number of columns (986) and the row names correspond to the number of rows (2782). However, the length of the classes (4) corresponds to the various components being a special table dataframe, table dataframe, table, and a data frame.


### Step 3: Inspect SNPs
View the data file:

```{r}
View(snp)
```
In viewing the data file, we can see that it is a table with headers. 

This file is currently a list:
```{r}
typeof(snp)
```
This file has 15 variables, with 983 rows and 15 columns of data.
```{r}
length(snp)
dim(snp)
```
By first storing the attributes of the snps file as an object, I can further analyze each attribute of the list.  
```{r results=FALSE}
snp_attributes <- attributes(snp)
length(snp_attributes$names)
length(snp_attributes$class)
length(snp_attributes$row.names)
typeof(snp_attributes$names)
typeof(snp_attributes$row.names)
typeof(snp_attributes$class)
snp_attributes$class
```
We can see that the lengths of each attribute, how the names correspond to the number of columns (15) and the row names correspond to the number of rows (983). However, the length of the classes (4) corresponds to the various components being a special table dataframe, table dataframe, table, and a data frame.


## Part II: Data Processing
### Step 1: Prep files
Sort the SNPs by ID
```{r}
library(tidyr)
library(tidyverse)
sorted_snp<-arrange(snp, SNP_ID)
```

### Step 1: Define groups
Pull out Maize (ZMMIL, ZMMLR, and ZMMMR)
```{r}
maize<- filter(fang, Group %in% c("ZMMIL", "ZMMLR", "ZMMMR"))
```
Pull out Teosinte (ZMPBA, ZMPIL, ZMPJA)
```{r}
teosinte<- filter(fang, Group %in% c("ZMPBA", "ZMPIL", "ZMPJA"))
```

### Step 2: Transpose file Maize and Teosinte
```{r}
Tmaize<- as.data.frame(t(maize))
Tteosinte <-as.data.frame(t(teosinte))
```

### Step 3: Join SNP and Genotype files for both Maize and Teosinte
```{r}

```


## Part III: Data Visualization

### SNPs per chromosom
### Missing data and amount of heterozygosity
### "Your own visualization
